tests/test_app.rs

```diff
@@ -140,6 +140,220 @@ impl Responder for Pet {
     }
 }
 
+#[test]
+fn test_path_order() {
```

Comment:
I love this test! Does it fail without the proposed changes?

Reply:
Yes it does.
Though after writing it I found out we also have another one: test_params, which was not failing when it should have! but it's also a more complicated test.